### PR TITLE
Move project player addons to a top-level category

### LINKED ISSUE #3014: Separate project player addons from editor addons
### Is your request related to a problem? Please describe.

On the settings page, the project player category is a subcategory of Scratch Editor Features, but the project player is a part of both the website and the editor.

### Describe the solution you'd like

Make it a separate top-level category.

### PR DESCRIPTION
**Pull request by Samq64**
Resolves #3014

### Changes

Moves project player addons to a new top-level category under the Scratch Editor Features category. Since other the other top-level category tags are short, the `projectPlayer` tag has been renamed to just `player`.

### Reason for changes

It's confusing for project player addons to be a subcategory of editor addons since those addons also affect the project page and embeds.

### Tests
Tested on Edge 96.

### CODE REVIEW

File: webpages/settings/data/tags.js [hunk 1]
```diff
@@ -44,6 +44,14 @@ export default [
       theme: true,
     },
   },
+  {
+    name: "forPlayer",
+    matchName: "player",
+    color: "darkgreen", // TODO: use a new tag color
```
Comment (RedGuy12) [inline webpages/settings/data/tags.js:50]: reminder

**Comment by WorldLanguages** [comment]
@mxmou Do you like this? I'm not convinced. (asking you since you made the issue that suggested this)

File: webpages/settings/data/tags.js (hunk 1 above)
Comment (Samq64) [reply webpages/settings/data/tags.js:50]: I'll do that later today or tomorrow.

**Comment by Samq64** [comment]
There's only one theme in the project player themes, so maybe it would be better to get rid of the project player themes category completely.

**Comment by WorldLanguages** [comment]
Honestly I don't really like this. IMO the project player is part of the editor.

**Comment by mxmou** [comment]
> Honestly I don't really like this. IMO the project player is part of the editor.

It's also part of the website.

**Comment by apple502j** [comment]
Manifest: ScratchAddons/manifest-schema#56

**Comment by TheColaber** [comment]
In my opinion, I agree with this being a separate category. The player which can be found in 3 places (projects/*/, projects/*/editor, projects/*/embed). I don't think that to change something in an embed, you have to look under the editor section. And @mxmou argued that this could technically also be a website addon since you can find it there too. I think that when most people think of an "editor" addon, they think of scripts that can run on projects/*/editor. However, all of these scripts that run, run on "projects", which are all 3, project page, project editor, and project embed. This is why I think they should have their own category.

**Comment by WorldLanguages** [comment]
I'm not convinced this should be merged, but I was more skeptical a few months ago.

Some issues I see:
1. There's less than 10 `projectPlayer` addons. And I'm not sure if clones and mouse-pos should be counted, because they only appear when inside the editor, and aren't helpful when viewing someone else's project.
2. The "Scratch Editor Features" would have 3 subcategories instead of four, making it less consistent with Scratch Website Features.
3. The sidebar of the settings page is already filled on low resolution screens. Adding one more button would hide the "more settings" button for some users.

**Comment by mxmou** [comment]
> 1. There's less than 10 `projectPlayer` addons. And I'm not sure if clones and mouse-pos should be counted, because they only appear when inside the editor, and aren't helpful when viewing someone else's project.

Clones and mouse-pos are editor addons. I don't think this is an issue - there aren't many popup addons either.

> 2. The "Scratch Editor Features" would have 3 subcategories instead of four, making it less consistent with Scratch Website Features.

Why is that important? More subcategories can be added later if there are enough addons for the same part of the editor.

> 3. The sidebar of the settings page is already filled on low resolution screens. Adding one more button would hide the "more settings" button for some users.

The category buttons are very big - reducing all of them to the size of "small" ones (Credits, Help Translate, Send Feedback) would add enough space for adding another button. The header could be slightly smaller too.

**Comment by WorldLanguages** [comment]
Sorry, I forgot to reply.

> Clones and mouse-pos are editor addons. I don't think this is an issue - there aren't many popup addons either.

I think I created the popup category because having "message count in extension icon" as a website addon was too weird. It did involve the Scratch website in a way, because the message count is taken from there, but in no way it modifies the Scratch website and it doesn't even link to it.
Having project player addons inside editor addons isn't as weird as that.

> Why is that important? More subcategories can be added later if there are enough addons for the same part of the editor.

Okay, good point. We have tags such as `comments` but they aren't shown as different subcategories because there's not enough addons with those tags.

> The category buttons are very big - reducing all of them to the size of "small" ones (Credits, Help Translate, Send Feedback) would add enough space for adding another button. The header could be slightly smaller too.

That could work.

**Comment by WorldLanguages** [comment]
Am I missing any addon that could fit into this category? I previously said clones and mouse-pos are targeted for developing a project, not for watching projects, so those don't really fit. Also, please see what I say at the end about remove-curved-stage-border.

- 60fps
- fullscreen
- gamepad
- mute-project
- pause
- remove-curved-stage-border (would this addon stay as a theme? would we need to create a "themes for project player" subcategory? this one is very confusing)

File: webpages/settings/data/tags.js (hunk 1 above)
Comment (Samq64) [reply webpages/settings/data/tags.js:50]: I think it's been decided there'll be no theme sub-category, so I'll just remove the tag completely.

**Comment by Samq64** [comment]
Done. By the way, the project player themes sub-category already existed, but I removed it.

**Comment by mxmou** [comment]
Added "pending" because the spacing of the category selector needs to be reduced first.

File: webpages/settings/data/tags.js (hunk 1 above)
Comment (Samq64) [reply webpages/settings/data/tags.js:55]: I re-added the theme category, but now #4989 is open and if it gets merged, the colour of all theme tags will be the same.

**Comment by Samq64** [comment]
Is this still pending discussion or can the pending label be removed since the padding has been reduced?

**Comment by mxmou** [comment]
@Samq64 The pending label was for padding.

**Comment by GrahamSH-LLK** [comment]
Is this still under development?

**Comment by Samq64** [comment]
I don't think we got to a conclusion on if this will be accepted or not.

**Comment by WorldLanguages** [comment]
There might have been a misunderstanding here: (does this PR remove the padding on the buttons or not?)
image

**Comment by mxmou** [comment]
@WorldLanguages It does reduce the padding values in `category-selector.html`.

**Comment by Samq64** [comment]
> There might have been a misunderstanding here: (does this PR remove the padding on the buttons or not?)

Yes it does: https://github.com/ScratchAddons/ScratchAddons/pull/3955/commits/122998b1ef50d16325f3c66add83bd71d90b4146

**Review by DNin01 — CHANGES_REQUESTED** [review]
If the plan is to make this change, we will need to update this pull request so the latest addons such as `vol-slider` are categorized correctly.

**Review by DNin01 — APPROVED** [review]
I think we're all set.

File: addons/clones/addon.json [hunk 2]
```diff
@@ -47,6 +47,6 @@
     "version": "1.31.0",
     "newSettings": ["projectpage"]
   },
-  "tags": ["editor", "projectPlayer", "recommended"],
+  "tags": ["editor", "recommended"],
```
Comment (iqnite) [inline addons/clones/addon.json:50]: Clone counter works on the website, too.
```suggestion
  "tags": ["player", "recommended"],
```

File: addons/mouse-pos/addon.json [hunk 3]
```diff
@@ -21,6 +21,6 @@
       "matches": ["projects"]
     }
   ],
-  "tags": ["editor", "projectPlayer", "recommended"],
+  "tags": ["editor", "recommended"],
```
Comment (iqnite) [inline addons/mouse-pos/addon.json:24]: Same here. You should consider that someone might add website support in the future.
```suggestion
  "tags": ["player", "editor", "recommended"],
```

**Review by iqnite — CHANGES_REQUESTED** [review]
Looks good! Could you take a look at the comments I've added? I don't know if it's intentional, but the addons that already had the "projectPlayer" tag shouldn't lose the "player" tag, because users might look for them in the "Player" category and wouldn't find them. Remember that addons can belong to multiple categories.

File: addons/mouse-pos/addon.json (hunk 3 above)
Comment (Samq64) [reply addons/mouse-pos/addon.json:24]: According to the [addon manifest docs](https://scratchaddons.com/docs/reference/addon-manifest/#required-factual-tags) addons can only have one top-level category tag.

File: addons/clones/addon.json (hunk 2 above)
Comment (Samq64) [reply addons/clones/addon.json:50]: It does now, but I think it's disabled by default outside the editor. I agree it should have the tag though.

File: addons/mouse-pos/addon.json (hunk 3 above)
Comment (iqnite) [reply addons/mouse-pos/addon.json:24]: > According to the [addon manifest docs](https://scratchaddons.com/docs/reference/addon-manifest/#required-factual-tags) addons can only have one top-level category tag.

Really? Oh, sorry then.

Comment (iqnite) [reply addons/mouse-pos/addon.json:24]: Clone counter still needs the `player` tag.

**Review by iqnite — APPROVED** [review]
Neat!

**Comment by Samq64** [comment]
This is still on the v1.36 milestone but Vue 3 is on v1.38. Is that the case or is this one also being pushed back to 1.38?

**Comment by Samq64** [comment]
Is this still planned for 1.37? Both Mxmou and DNin previously approved.

**Comment by WorldLanguages** [comment]
Hmm, I'll be looking at this after the MV3 release. Sorry.

**Review by mxmou — APPROVED** [review]

**Comment by Samq64** [comment]
@Hans5958 For the manifest schema does it need to be a new version or do I edit the existing one?

**Comment by DNin01** [comment]
> @Hans5958 For the manifest schema does it need to be a new version or do I edit the existing one?

I'm pretty sure we should bump the schema to a new version. The last schema hasn't changed for a few releases.

**Pull request merged**